libraries/standard/http/src/http_client.c

```diff
@@ -247,40 +308,203 @@ static HTTPStatus_t _sendHttpBody( const HTTPTransportInterface_t * pTransport,
     HTTPStatus_t returnStatus = HTTP_SUCCESS;
     int32_t transportStatus = 0;
 
-    /* Send the HTTP body over the network. */
-    if( pRequestBodyBuf != NULL )
+    assert( pTransport != NULL );
+    assert( pTransport->send != NULL );
+    assert( pRequestBodyBuf != NULL );
+
+    transportStatus = pTransport->send( pTransport->pContext,
+                                        pRequestBodyBuf,
+                                        reqBodyBufLen );
+
+    if( transportStatus < 0 )
     {
-        transportStatus = pTransport->send( pTransport->pContext,
-                                            pRequestBodyBuf,
-                                            reqBodyBufLen );
+        IotLogErrorWithArgs( "Failed to send HTTP body: Transport send() "
+                             " returned error: TransportStatus=%d",
```

Comment:
We'll want to find a way to convert `transportStatus` to a string for logging here, otherwise we're going to have 1000 issues that say _"I received error -119, what does that mean?"_  Ideally, the transport layer should have a function for converting error codes to strings since every transport layer will do things differently.

Reply:
That will be part of the demo since the transportStatus is unique to each stack.